core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -51,20 +52,21 @@ var CollectionEditorPage = function() {
   this.addExistingExploration = function(explorationId) {
     addExplorationInput.sendKeys(explorationId);
     // Waits until the button becomes active after debouncing.
-    browser.wait(until.elementToBeClickable(addExplorationInput), 5000,
-      'Unable to find exploration Id ' + explorationId);
+    waitFor.elementToBeClickable(
+      addExplorationButton,
+      'Unable to find exploration ID: ' + explorationId);
     addExplorationButton.click();
   };
 
   // Search and add existing exploration to the node graph.
   this.searchForAndAddExistingExploration = function(query) {
     browser.wait(until.visibilityOf(addExplorationInput), 10000,
-      'Add Exploration Input is not visible').then(function(isVisbile) {
-      if (isVisbile) {
+      'Add Exploration Input is not visible').then(function(isVisible) {
+      if (isVisible) {
         addExplorationInput.sendKeys(query);
         // Need to wait for result to appear.
-        browser.wait(until.elementToBeClickable(addExplorationInput), 5000,
-          'Unable to find exploration ' + query);
+        waitFor.elementToBeClickable(
+          addExplorationInput, 'Unable to find exploration: ' + query);
       }
```

Comment:
This expected condition should be for `addExplorationButton` instead of `addExplorationInput`. The  "Add Exploration" button won't be clickable if there is zero result from finding an exploration that has a matching title or objective with the query.

Reply:
By the time "Add Exploration" button get checked for "isEnable" down below, this expected condition should already fail!

Reply:
Done. Thanks!